pytensor/link/numba/dispatch/slinalg.py

```diff
@@ -155,7 +155,7 @@ def impl(A, B, trans, lower, unit_diagonal, b_ndim, overwrite_b):
             DIAG,
             N,
             NRHS,
-            np.asfortranarray(A).T.view(w_type).ctypes,
+            A_copy.view(w_type).ctypes,
```

Comment:
Was this the bug? What was the old logic trying to achieve?

Also what does this `view(wtype).ctypes` do exactly?

Reply:
I honestly don't remember why I transposed this originally. I think I was thinking that since C-format transposed is F-format I should transpose? But that's dumb. 

`.view(w_type).ctypes` gets back a c-pointer for the original object. I copied the logic from [here](https://github.com/numba/numba/issues/5301). The `w_type` thing is to handle potentially complex inputs, since C has no complex datatype, you need a view into the "associated" float type (c128 -> w_type = f64, c64-> w_type = f32)